code/modules/events/stray_cargo.dm

```diff
@@ -7,13 +7,45 @@
 	earliest_start = 10 MINUTES
 	category = EVENT_CATEGORY_BUREAUCRATIC
 	description = "A pod containing a random supply crate lands on the station."
+	admin_setup = /datum/event_admin_setup/stray_cargo
+
+/datum/event_admin_setup/stray_cargo
+	///Admin setable override that is used instead of selecting a random location
+	var/atom/landing_turf_override
+	///Admin setable override to spawn a specific cargo pack type
+	var/pack_type_override
+
+/datum/event_admin_setup/stray_cargo/prompt_admins()
+	var/admin_targeted = tgui_alert(usr,"Aimed at current turf?", "Pod Targetting", list("Yes", "No"))
+	if(admin_targeted == "Yes")
+		landing_turf_override = get_turf(usr)
+	else
+		landing_turf_override = null
+	var/admin_selected_pack = tgui_alert(usr,"Select pod contents?", "Pod Contents", list("Yes", "No"))
+	if(admin_selected_pack == "Yes")
+		override_contents()
+	else
+		pack_type_override = null
+	message_admins("[key_name_admin(usr)] has aimed a stray cargo pod at [AREACOORD(landing_turf_override)]. The pod contains [pack_type_override].")
+	log_admin("[key_name_admin(usr)] has aimed a stray cargo pod at [AREACOORD(landing_turf_override)]. The pod contains [pack_type_override].")
```

Comment:
If both landing_turf_override/pack_type_override is null (both cases you deliberately set right before this log is sent off), won't you get a message like:

"NamelessFairy has aimed a stray cargo pod at . The pod contains ."

Do account for that when the message to log is created, probably via in-string nullcheck or just simply rework how you "build" the string? Up to you, but should definitely be watchful for.

Reply:
Fixed, this is mostly a holdover from how rods do admin setup, basically treating null as "random" it makes sense but its probably not the best way of doing things.